File: PeriodisationProgramApp.Domain/Entities/TrainingProgram.cs
using PeriodisationProgramApp.Domain.Enums;

namespace PeriodisationProgramApp.Domain.Entities
{
    public class TrainingProgram : CommunityEntity<UserTrainingProgramLike, UserTrainingProgramRating>
    {
        public string? Name { get; set; }

        public string? Description { get; set; }

        public List<TrainingSession> Sessions { get; set; } = new();

        public TrainingProgramType Type { get; set; }

        public TrainingLevel TrainingLevel { get; set; }

        public int NumberOfSessions { get; set; }
    }
}

File: PeriodisationProgramApp.Web/src/types/enitities/TrainingProgram.ts
import { TrainingLevel } from "../../enums/TrainingLevel";
import { TrainingProgramType } from "../../enums/TrainingProgramType";
import { CommunityEntity } from "./CommunityEntity";
import { TrainingSession } from "./TrainingSession";

export type TrainingProgram = {
  name: string;
  description?: string;
  type: TrainingProgramType;
  numberOfSessions: number;
  trainingLevel: TrainingLevel;
  sessions: TrainingSession[];
} & CommunityEntity;
